compiler/codegen/cg-expr.cpp

```diff
@@ -6116,7 +6139,13 @@ DEFINE_PRIM(DYNAMIC_CAST) {
 DEFINE_PRIM(STACK_ALLOCATE_CLASS) {
     AggregateType* at = toAggregateType(call->get(1)->typeInfo());
     const char* struct_name = at->classStructName(true);
-    GenRet tmp = createTempVar(struct_name);
+    GenRet tmp = createTempVar(struct_name,
+#ifdef HAVE_LLVM
+      gGenInfo->cfile ? 0 : at->symbol->getLLVMStructureAlignment()
+#else
+      0  // ignored for C codegen
+#endif
+    );
```

Comment:
Can you make this code more readable? Maybe a temporary instead of doing it in-place like this?